src/providers/azure-ai-inference/api.ts

```diff
@@ -0,0 +1,63 @@
+import { ProviderAPIConfig } from '../types';
+
+const AzureAIInferenceAPI: ProviderAPIConfig = {
+  getBaseURL: ({ providerOptions }) => {
+    const {
+      azureDeploymentName,
+      azureRegion,
+      azureDeploymentType,
+      provider,
+      azureEndpointName,
+    } = providerOptions;
+    if (provider === 'github') {
+      return 'https://models.inference.ai.azure.com';
+    }
+    if (azureDeploymentType === 'serverless') {
+      return `https://${azureDeploymentName?.toLowerCase()}.${azureRegion}.models.ai.azure.com`;
+    }
+
+    return `https://${azureEndpointName}.${azureRegion}.inference.ml.azure.com/score`;
+  },
+  headers: ({ providerOptions }) => {
+    const { apiKey, azureDeploymentType, azureDeploymentName } =
+      providerOptions;
+    const headers: Record<string, string> = {
+      Authorization: `Bearer ${apiKey}`,
+      'extra-parameters': 'ignore',
+    };
+    if (azureDeploymentType === 'managed' && azureDeploymentName) {
+      headers['azureml-model-deployment'] = azureDeploymentName;
+    }
```

Comment:
https://github.com/Azure/azure-sdk-for-python/blob/31688d9ab355f96384dc3b2bdaee84a3b5c7f021/sdk/ml/azure-ai-ml/azure/ai/ml/operations/_local_endpoint_helper.py#L88-L89